.github/workflows/deploy-docs.yml

```diff
@@ -1,10 +1,13 @@
 name: Build and Deploy docs
 
 on:
+  pull_request:
```

Comment:
@ayushanand18 we want to run in all PRs and merges to main to ensure the docs are building but we do not want to publish but we only want to publish when releasing.

Reply:
Gotcha! I understand the workflow now 👀

Reply:
It took me a while to understand all that. I created this problem in the original PR BTW :flushed: